internal/compiler/transforms/compile/css-handler/prefix/utils.ts

```diff
@@ -0,0 +1,246 @@
+import {
+	CSSBlock,
+	CSSDeclaration,
+	cssBlock,
+	cssDeclaration,
+	cssIdentifier,
+} from "@internal/ast";
+import {UnknownObject} from "@internal/typescript-helpers";
+import {
+	CompilerPath,
+	EnterSignal,
+	ExitSignal,
+	createVisitor,
+	signals,
+} from "@internal/compiler";
+import {
+	VisitorStateEnter,
+	VisitorStateExit,
+} from "@internal/compiler/lib/VisitorState";
+import {Browser} from "@internal/browser-features/Browser";
+
+export function nodeHasPrefixedProperty(
+	node: CSSBlock,
+	property: string,
+	prefix: string,
+	rename: (propertyName: string) => string = (propertyName) => propertyName,
+): boolean {
+	if (node.value) {
+		return node.value.some((n) =>
+			n.type === "CSSDeclaration" && n.name === rename(`-${prefix}-${property}`)
+		);
+	}
+	return false;
+}
+
+export function nodeHasPrefixedPropertyValue(
+	node: CSSBlock,
+	property: string,
+	value: string,
+	prefix: string,
+	rename: (value: string) => string = (value) => value,
+): boolean {
+	if (node.value) {
+		return node.value.some((n) =>
+			n.type === "CSSDeclaration" &&
+			n.name === property &&
+			n.value.length === 1 &&
+			n.value[0]?.type === "CSSIdentifier" &&
+			n.value[0].value === rename(`-${prefix}-${value}`)
+		);
+	}
+	return false;
+}
+
+export function nodePropertyIndex(node: CSSBlock, property: string): number {
+	return node.value
+		? node.value.findIndex((n) =>
+				n.type === "CSSDeclaration" && n.name === property
+			)
+		: -1;
```

Comment:
Not a fan of multiline ternaries, at least how we format this one:

```suggestion
	if (node.value === undefined) {
		return node.value.findIndex((n) =>
			n.type === "CSSDeclaration" && n.name === property
		);
	} else {
		return -1;
	}
```